script.js

```diff
@@ -1,12 +1,286 @@
+  showSummary.innerHTML = `${show.summary}`;
+  showGenres.innerText = `Genres: ${show.genres
+    .toString()
+    .replace(/,/g, " | ")}`;
+  showStatus.innerHTML = `Status: ${show.status}`;
+  showRating.innerHTML = `Rated: ${show.rating.average}`;
+  showRuntime.innerHTML = `Runtime: ${show.runtime} minutes`;
+
+  showCard.appendChild(showName);
+  showCard.appendChild(showImage);
+  showCard.appendChild(showSummary);
+  showCard.appendChild(showGenres);
+  showCard.appendChild(showStatus);
+  showCard.appendChild(showRating);
+  showCard.appendChild(showRuntime);
+  showContainer.appendChild(showCard);
+
+  showName.addEventListener("click", getEpisodes);
+}
+
+function getEpisodes(event) {
+  let showId;
+  let showName = event.target.textContent;
+  allShows.forEach((show) => {
+    if (show.name.includes(showName)) {
+      showId = show.id;
+      return showId;
+    }
+    showContainer.style.display = "none";
+    showSearchBar.style.display = "none";
+    searchBar.style.display = "initial";
+    selector.innerHTML = "";
+    selector.innerHTML = "<option value ''>--select episode--</option>";
+    selector.value = "";
+  });
+
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data;
+
+      makePageForEpisodes(allEpisodes);
+      selector.addEventListener("change", selectedEpisode);
+      selector.style.visibility = "visible";
+    })
+    .catch((error) => console.log("Error:", error));
+}
+
+backButton.addEventListener("click", goBackToMainPage);
+
+function goBackToMainPage() {
+  makePageForShows(allShows);
+  selector.value = "";
+  showSelector.value = "";
+  selector.style.visibility = "hidden";
+  container.innerHTML = "";
+  showContainer.style.display = "initial";
+  showSearchBar.style.display = "initial";
+  searchBar.style.display = "none";
+  searchBar.value = "";
+  showSearchBar.value = "";
+  displayShowCounter();
+}
+
+function makePageForShows(showList) {
+  showContainer.innerHTML = "";
+  showCounter = 0;
+  showList.forEach((show) => {
+    renderShowListing(show);
+    showCounter++;
+    displayShowCounter();
+  });
+}
+
+function displayShowCounter() {
+  heading.innerHTML = `Displaying ${showCounter}/${allShows.length}`;
+}
+
+function episodeIsIncluded(episode, searchBarValue) {
+  if (searchBarValue === undefined) {
+    return true;
+  }
+  return (
+    episode.name.toLowerCase().includes(searchBarValue.toLowerCase()) ||
+    episode.summary.toLowerCase().includes(searchBarValue.toLowerCase())
+  );
+}
+
+function makeShowDropDownList(showsList) {
+  showsList.forEach((show) => {
+    const showOption = document.createElement("option");
+    showOption.className = "show-options";
+    const showId = (showOption.value = show.id);
+
+    showOption.innerHTML = show.name;
+
+    showSelector.appendChild(showOption);
+
+    showSelector.addEventListener("change", selectedShow);
+    function selectedShow() {
```

Comment:
This is great function. Easy to read and accounts for errors. Astounded by the level of work you have managed to produce.